### DISCUSSION TITLE
What the best way to collect custom metrics of my hydra instanse?

### DISCUSSION THREAD

**Discussion by cat-in-vacuum** [ory/hydra#2654] [Q&A] [answered]
Hello! I need a little help to reaching the observability of my hydra instance.
The current goal is to get more metrics than existing from the box like a prediction of expired tokens on the future period (e.t. in 1 hour or one day) , count of success/fail grants of refresh/access tokens, grouping request routes by response codes.
I looked a guide and looked source code but don't find anything regard to these things
For now it seems like I have only one approach  - get such metrics from the database. But it looks like a dirty hack.  Ideally, I would like to get it on the fly
Help me please to resolve the issue or share how you handle doing with it

**Comment by Benehiko** [comment, marked as answer]
Hi @cat-in-vacuum,

This looks similar to our [Distributed Tracing](https://www.ory.sh/hydra/docs/guides/tracing) support on Hydra. Of course it won't contain everything and would require you to pull data directly from the database if you need other custom metrics.

**Reply by vinckr** [reply]
Hey @cat-in-vacuum , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
